Sources/PostgresNIO/New/PSQLConnection.swift

```diff
@@ -203,7 +200,7 @@ final class PSQLConnection {
         logger[postgresMetadataKey: .connectionID] = "\(connectionID)"
         
         return eventLoop.flatSubmit {
-            eventLoop.makeSucceededFuture(Void()).flatMapThrowing { _ -> SocketAddress in
+            eventLoop.submit { () throws -> SocketAddress in
```

Comment:
(this suggestion actually applies to the next 38 lines of code, but GH won't let me apply this comment to most of it)
```suggestion
            let address: SocketAddress
            
            switch configuration.connection {
            case let .resolved(addr, _): address = addr
            case let .unresolved(host, port): address = try .makeAddressResolvingHost(host, port: port)
            }
            let bootstrap = ClientBootstrap(group: eventLoop)
                .channelInitializer { channel in
                    let decoder = ByteToMessageHandler(PSQLBackendMessage.Decoder())
                    let enableSSLCallback = configuration.tlsConfiguration.flatMap { tlsConfiguration in { channel in
                        channel.eventLoop.submit {
                            let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
                            return try NIOSSLClientHandler(
                                context: sslContext,
                                serverHostname: configuration.sslServerHostname)
                            }.flatMap { sslHandler in
                                channel.pipeline.addHandler(sslHandler, position: .before(decoder))
                            }
                        }
                    }
                        
                    return channel.pipeline.addHandlers([
                        decoder,
                        MessageToByteHandler(PSQLFrontendMessage.Encoder(jsonEncoder: configuration.coders.jsonEncoder)),
                        PSQLChannelHandler(
                            authentification: configuration.authentication,
                            logger: logger,
                            enableSSLCallback: enableSSLCallback),
                        PSQLEventsHandler(logger: logger)
                    ])
                }
                
            return bootstrap
                .connect(to: address)
            .flatMap { channel -> EventLoopFuture<Channel> in
```